Essentials/UI/CooldownButton.cs

```diff
@@ -1,295 +1,226 @@
+        }
 
+        private void Start()
+        {
+            killButtonManager = UnityEngine.Object.Instantiate(hudManager.KillButton, hudManager.transform);
+            startColorButton = killButtonManager.renderer.color;
+            startColorText = killButtonManager.TimerText.Color;
+            killButtonManager.gameObject.SetActive(true);
+            killButtonManager.renderer.enabled = true;
             Texture2D tex = GUIExtensions.CreateEmptyTexture();
-            byte[] buttonTexture = Reactor.Extensions.Extensions.ReadFully(resourceStream);
-            ImageConversion.LoadImage(tex, buttonTexture, false);
-            KillButtonManager.renderer.sprite = GUIExtensions.CreateSprite(tex);
-
-            PassiveButton button = KillButtonManager.GetComponent<PassiveButton>();
+            ImageConversion.LoadImage(tex, this.image, false);
+            killButtonManager.renderer.sprite = GUIExtensions.CreateSprite(tex);
+            PassiveButton button = killButtonManager.GetComponent<PassiveButton>();
             button.OnClick.RemoveAllListeners();
-            button.OnClick.AddListener(new Action(() =>
+            button.OnClick.AddListener((UnityEngine.Events.UnityAction)listener);
+            void listener()
             {
-                if (Timer < 0f && CanUse)
+                if (Timer < 0f && canUse)
                 {
-                    KillButtonManager.renderer.color = new Color(1f, 1f, 1f, 0.3f);
-
+                    killButtonManager.renderer.color = new Color(1f, 1f, 1f, 0.3f);
                     if (hasEffectDuration)
                     {
-                        IsEffectActive = true;
-
+                        isEffectActive = true;
                         Timer = EffectDuration;
-
-                        KillButtonManager.TimerText.Color = new Color(0, 255, 0);
+                        killButtonManager.TimerText.Color = new Color(0, 255, 0);
                     }
-
-                    onClick();
+                    OnClick();
                 }
-            }));
-        }
-
-        /// <summary>
-        /// Call in HudManager.Start
-        /// </summary>
-        public CooldownButton(Action onClick, float cooldown, string imageEmbededResourcePath, Vector2 positionOffset, ButtonCategory category, float effectDuration, Action onEffectEnd) : this(Assembly.GetCallingAssembly(), onClick, cooldown, imageEmbededResourcePath, positionOffset, category, true, effectDuration, onEffectEnd)
-        {
+            }
         }
-
-        /// <summary>
-        /// Call in HudManager.Start
-        /// </summary>
-        public CooldownButton(Action onClick, float cooldown, string imageEmbededResourcePath, Vector2 positionOffset, ButtonCategory category) : this(Assembly.GetCallingAssembly(), onClick, cooldown, imageEmbededResourcePath, positionOffset, category, false, 0F, null)
+        public bool CanUse()
         {
+            if (PlayerControl.LocalPlayer.Data == null) return false;
```

Comment:
`PlayerControl.LocalPlayer` can also be null, resulting an exception, to check for both, use `PlayerControl.LocalPlayer?.Data`.